check-licenses/action.yml

```diff
@@ -0,0 +1,68 @@
+name: "Check licenses"
+description: "Check if package licenses in the current environment are among the accepted ones"
+
+inputs:
+  python-version:
+    description: 'Desired Python version.'
+    default: '3.10'
+    required: false
+    type: string
+  use-python-cache:
+    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
+    required: false
+    default: true
+    type: boolean
+  skip-install:
+    description: 'Skip project installation.'
+    default: false
+    required: false
+    type: boolean
+
+runs:
+  using: "composite"
+  steps:
+    - name: "Install Git and clone project"
+      if: inputs.skip-install == 'false'
+      uses: actions/checkout@v3
+
+    - name: "Set up Python"
+      if: inputs.skip-install == 'false'
+      uses: pyansys/actions/_setup-python@main
+      with:
+        python-version: ${{ inputs.python-version }}
+        use-cache: ${{ inputs.use-python-cache }}
+
+    - name: "Update pip"
+      if: inputs.skip-install == 'false'
+      shell: bash
+      run: python -m pip install -U pip
+
+    - name: "Install Python library"
+      if: inputs.skip-install == 'false'
+      shell: bash
+      run: python -m pip install .
+
+    - name: "Check licences of packages"
+      shell: bash
+      run: |
+        python -m pip install pip-licenses
+        wget https://raw.githubusercontent.com/pyansys/actions/f06b839725de78136d4f797b95d5c6458c689cd3/check-licenses/accepted-licenses.txt
+        wget https://raw.githubusercontent.com/pyansys/actions/f06b839725de78136d4f797b95d5c6458c689cd3/check-licenses/ignored-packages.txt
```

Comment:
Note that these URLs need are currently pointing to the files in the branch you created, @GuillemBarroso. Once everything is ready to be merged, they should be changed assuming that the files are in `main`. This will make the actions to fail in this PR but pass once merged.

Reply:
Also, could we isolate the download of the files in a separate step? Just to split de setup process and the final logic of the licensing checking.

Reply:
Oh true! Yeah, those URLs have to be modified =)

And yeah, we might want to break down the entire step into more isolated ones. Good idea @jorgepiloto.

Reply:
@jorgepiloto, yes. This applies to both the links in ``action.yml`` and the @feat/chech-licenses in ``ci_cd.yml``.

What's the proper way of doing that? I was thinking on first merging the PR (should pass since the branch will still be "alive"), and then proposing another PR changing the two files mentioned above to point to @main with the correct links.

Finally, I think the links in ``action.yml`` have to be updated every time that the files are modified. Is this correct? (For example, now the PR in pyanys/pyansys is failing because it does not download the updated version of the files).